framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3106,7 +3106,7 @@ public boolean validateTypeOf(Tree tree) {
             }
             break;
         default:
-            type = atypeFactory.getAnnotatedType(tree);
+        	type = atypeFactory.getDefaultedAnnotatedType(tree);
```

Comment:
I agreed that getDefaultedAnnotatedType should be used to validate the LHS of an assignment, but this code uses it for all expressions.  Given my proposed changes to getDefaultedAnnotatedType (#623), a test case is absolutely necessary to prevent a regression.  

I tried to reproduce the error with the PolyAllChecker (framework/tests/poly-all/polyall).
````
import polyall.quals.H1S1;
import polyall.quals.H1S2;
import polyall.quals.H1Top;

public @H1S2 class NewClass {
    public @H1S2 NewClass(){}
    void foo() {
        @H1S2 NewClass classA = new @H1Top NewClass();
        @H1S2 NewClass classA2 = classA;
        @H1S1 NewClass classA3;
    }
}
````

````
javacheck -cp ../../tests/build/ -processor polyall.PolyAllChecker NewClass.java
NewClass.java:8: error: [type.invalid] [@H1Top, @H2Top] may not be applied to the type "@H1Top @H2Top NewClass"
        @H1S2 NewClass classA = new @H1Top NewClass();
                                ^
NewClass.java:10: error: [type.invalid] [@H1S1, @H2Top] may not be applied to the type "@H1S1 @H2Top NewClass"
        @H1S1 NewClass classA3;
                       ^
2 errors
````
But there is no error on the refined type of classA.  So I'm not sure why the Universe type system is issuing an error.